common/persistence/sql/sqlplugin/mysql/session/session.go

```diff
@@ -145,7 +196,13 @@ func buildDSNAttrs(cfg *config.SQL) map[string]string {
 		attrs[k] = v
 	}
 
-	return attrs
+	if version == MySQLVersion8_0 && dbKind == sqlplugin.DbKindVisibility {
```

Comment:
Why do you need the version check?
I'm removing support for MySQL 5.7 [here](https://github.com/temporalio/temporal/pull/5387), so this check will be gone soon.

Reply:
Oh nice! And I added this because mysql 5.7 worked just fine with this config enabled

Reply:
It worked because MySQL 5.7 doesn't support advanced visibility which is where the issue is.